src/app/models/operation.model.ts

```diff
@@ -1,43 +1,44 @@
+import {Node} from "./node.model";
 import {Trainrun} from "./trainrun.model";
-import {TrainrunSection} from "./trainrunsection.model";
 
 enum OperationType {
   create = "create",
   update = "update",
-  delete = "delete"
+  delete = "delete",
 }
 
-export abstract class Operation {
+enum OperationObjectType {
+  trainrun = "trainrun",
+  node = "node",
+};
+
+abstract class Operation {
   readonly type: OperationType;
+  readonly objectType: OperationObjectType;
+  readonly object: Trainrun | Node;
```

Comment:
Ah, when I was talking about dropping the generic, my intention was to also drop this field and move it to the more specialized operation classes :P

Reply:
I got it (i guess :D), done